src/material-experimental/mdc-input/input.spec.ts

```diff
@@ -1081,7 +1081,7 @@ describe('MatMdcInput with forms', () => {
       expect(describedBy).toBe(errorIds);
     }));
 
-    it('should not set `aria-invalid` to true if the input is empty', fakeAsync(() => {
+    it('should set `aria-invalid` to true if the input is empty', fakeAsync(() => {
```

Comment:
Note:  This test checks that a non-`required` `input` with a `FormControl` which contains `Validators.required` sets `aria-invalid`.

This actually is the edge case that @zelliott pointed out in #22779 when they mentioned that the logic for setting `aria-invalid` should actually be `(empty && required) ? null : errorState`.

In this case there is no `required` on the form field, and therefore no `aria-required`.  This means we _must_ set `aria-invalid` to indicate to a screen reader that there is some other validation error on the form field that they need to be aware of.

I'm not certain that this test's title is informative enough in this case -- thoughts?